source/vecs/entitymanager.d

```diff
@@ -384,13 +385,14 @@ public:
 	Params:
 		entity = a valid entity.
 	*/
-	@system
-	void removeAllComponents(in Entity e)
+	void removeAllComponents()(in Entity e)
 		in (validEntity(e))
 	{
+		import std.traits : functionAttributes, SetFunctionAttributes;
+
 		foreach (sinfo; storageInfoMap)
 			if (sinfo.storage !is null)
-				sinfo.remove(e);
+				(() @trusted => cast(SetFunctionAttributes!(typeof(sinfo.remove), "D", functionAttributes!Fun)) sinfo.remove)()(e);
```

Comment:
Isn't this risky or the type is enforced somewhere else?

Reply:
This is needed. In `Storage` the attributes of `remove` are auto inferred by the `delegate` provided. If you look at the implementation of `removeComponent` this isn't needed because we can access `Storage` directly, so we can infer the template with the original attributes from `Storage`, which are the attributes of the `delegate` provided. The same is not possible here. We have to iterate all `StorageInfo`. This structure does not know nor care what Component type the storage has. So the only way to access the `Storage` method is through the delegates defined at its construction that point to the correct method. Because `remove` is templated it can have multiple attributes and a `delegate` signature must have predefined attributes. So the `delegate` mustn't have any attributes so that it can point to whatever `remove` function with any attribute combination. To ensure conformity with `Storage` in `EntityManager` I had to manually match the attributes, because `remove` from `StorageInfo` is `@system`.

TL/DR: this is safe to do

Reply:
> TL/DR: this is safe to do

Cool